waspc/cli/src/Wasp/Cli/Command/Requires.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Wasp.Cli.Command.Requires
+  ( -- * Using command requirements
+
+    -- This module exports the lower-level parts of the command requirement
+    -- API. To use these features to establish a requirement for a command,
+    -- use 'Wasp.Cli.Command.Common.commandRequires'. For convenience,
+    -- 'CommandRequirement' and its constructors are also exported from that
+    -- module.
+    --
+    -- Call @commandRequires SomeRequirementConstructor@ anywhere in your command,
+    -- preferably near the beginning of the command implementation and definitely
+    -- before using the requirement.
+    CommandRequirement (..),
+    RequiresT,
+    runRequiresT,
+    MonadRequires (..),
+  )
+where
+
+import Control.Monad.Error.Class (MonadError)
+import Control.Monad.IO.Class (MonadIO)
+import Control.Monad.State.Strict (StateT, evalStateT, gets, modify, when)
+import Control.Monad.Trans (MonadTrans)
+import Data.Foldable (find)
+
+-- | A requirement that a command needs in order to run
+data CommandRequirement
+  = -- | Requires that it is possible to connect to the prisma database.
+    DbConnection
+  deriving (Eq, Show)
+
+-- | A monad transformer that holds a list of satisfied 'CommandRequirement's
+-- that is updated whenever a new requirement is checked.
+--
+-- This is the canonical implementation of 'MonadRequires'.
+newtype RequiresT m a = RequiresT {unRequiresT :: StateT [CommandRequirement] m a}
```

Comment:
If you went for making RequiresT more general, than you could make it take a type parameter that represents the requirements, instead of it being hardcoded like this.